sstables/compaction_manager.cc

```diff
@@ -596,17 +596,28 @@ future<> compaction_manager::rewrite_sstables(column_family* cf, sstables::compa
     task->compacting_cf = cf;
     task->type = options.type();
     _tasks.push_back(task);
-    _stats.pending_tasks++;
 
-    task->compaction_done = repeat([this, task, options, get_func = std::move(get_func)] () mutable {
+    auto sstables = std::make_unique<std::vector<sstables::shared_sstable>>(get_func(*cf));
+    auto w = sstables.get();
```

Comment:
`w` is not a relevant name anymore.

Reply:
you are right, because now we no longer update the counters. Patch will be simpler too.

Let me spend some time to track down the failure in debug mode and then I'll resend what is hopefully the final version

Reply:
This is still not solved.